### PR TITLE
CCv0 | ci: Add CCv0 features into CI script

### PR DESCRIPTION
**Pull request by stevenhorsman**
Make the build understand if they are running against a target branch
of CCv0 or main and update branches dynamically based on this

Add code to get containerd from our CCv0 PR rather than a released build

Add new CCv0 condition to the code that created the kata image.
CCv0 requires some extra packages for running and debug. It also needs
umoci, which is not yet availabe as a distro package, so wget it direct

Fixes: #3887
Depends-on: github.com/kata-containers/kata-containers#2577

Signed-off-by: stevenhorsman <steven@uk.ibm.com>

### CODE REVIEW

File: .ci/install_cri_containerd.sh
Comment (jodh-intel) [inline .ci/install_cri_containerd.sh:32]: This looks like a temporary tweak which can be dropped when it gets merged by the upstream. If so, it might be safer to add a `do-not-merge` label to this PR. That way, we can still test if the code works in the Kata CI, but we won't land it by mistake.

Alternatively, if you preceive it may take a while to get that PR landed, are you happy to update this script once the PR does land?

Either way, I think it would be useful to add a call like the following to make this stand out in the CI output:

```bash
warn "Using patched Confidential Computing containerd version: see https://github.com/containerd/containerd/pull/5911"
```

(Note that it's best to spell out what "CC" refers to as for some of the "old hands", that acronym can also stand for https://github.com/clearcontainers :smile:)

**Review by jodh-intel — COMMENTED** [review]
Thanks @stevenhorsman. A few comments.

File: .ci/install_kata_image.sh
Comment (stevenhorsman) [reply .ci/install_kata_image.sh]: Thanks for the advice. I think I've updated this as required

File: .ci/install_cri_containerd.sh
Comment (stevenhorsman) [reply .ci/install_cri_containerd.sh:32]: Hey @jodh-intel, thanks very much for the review and suggestions. I've added the warn statement, but am a little unsure about the hard-coded version change. It's a bit of chicken and egg and I originally made the proper change in version.yaml, but this required an update in `kata-containers` repo that I couldn't test with this PR, so I was going to get this in, then update kata-containers, then come back to tests and fix, but that's not necessarily very efficient.
Alternatively I do have a PR open in that repo now (https://github.com/kata-containers/kata-containers/pull/2556), so I could put the version change in there, and make this change proper and depend on that?

**Review by jodh-intel — APPROVED** [review]
Thanks @stevenhorsman.

Either method is fine. We just need to ensure we don't forget to update this code and continue to rely on a patched version of containerd. You could prefix the warn message with "FIXME" possibly as that would help if we ever resurect the "FIXME checker" :smile:

File: .ci/install_cri_containerd.sh
Comment (stevenhorsman) [reply .ci/install_cri_containerd.sh:32]: As an update I decided to make the version change in https://github.com/kata-containers/kata-containers/pull/2556, so once that has been merged I'll update the code here to remove the TODO'd temporary code in this PR

Comment (stevenhorsman) [reply .ci/install_cri_containerd.sh:32]: As a further update, based on feedback from Samuel, I've separated the version change from the PR and created #2577 as a stand-alone PR with this change in now.

**Pull request merged**
